src/main/java/seedu/address/ui/UpcomingTuitionListPanel.java

```diff
@@ -77,33 +75,55 @@ public int compare(Tuition tuition1, Tuition tuition2) {
-        LocalDate weekToday = today.plusWeeks(1);
-        for (Tuition tuition : tuitionList) {
-            LocalDate sessionDate = tuition.getSession().getSessionDate().getDate();
-            if (today.compareTo(sessionDate) <= 0 && sessionDate.compareTo(weekToday) < 0) {
-                finalTuitionList.add(tuition);
-            }
-        }
-    }
-
-    /**
-     * Populates the tuitionList with {@code Tuition} for all sessions in the studentList.
+     * Populates the tuitionList with {@code Tuition} for all sessions in the studentList between today and
+     * the day after tomorrow.
      */
     private void populateTuitionList(ObservableList<Student> studentList, ObservableList<Tuition> tuitionList) {
+
+        SessionDate today = new SessionDate(LocalDateTime.now());
+        SessionDate tomorrow = new SessionDate(LocalDateTime.now().plusDays(1));
+        SessionDate dayAfterTomorrow = new SessionDate(LocalDateTime.now().plusDays(2));
+
         for (int i = 0; i < studentList.size(); i++) {
             Student currStudent = studentList.get(i);
             for (int j = 0; j < studentList.get(i).getListOfSessions().size(); j++) {
                 Session currSession = currStudent.getListOfSessions().get(j);
-                tuitionList.add(new Tuition(currStudent, currSession, i, j));
+                if (currSession instanceof RecurringSession) {
+                    RecurringSession recurringSession = (RecurringSession) currSession;
+
+                    if (!recurringSession.endBefore(today)) {
+                        if (recurringSession.hasSessionOnDate(today)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(today.getDate()), i, j));
+                        }
+                        if (recurringSession.hasSessionOnDate(tomorrow)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(tomorrow.getDate()), i, j));
+                        }
+                        if (recurringSession.hasSessionOnDate(dayAfterTomorrow)) {
+                            tuitionList.add(new Tuition(currStudent,
+                                    recurringSession.buildSessionOnDate(dayAfterTomorrow.getDate()), i, j));
+                        }
+                    }
+                } else {
+                    LocalDate dateOfSingleSession = currSession.getSessionDate().getDate();
+                    if (isWithinThreeDaysRange(today, dayAfterTomorrow, dateOfSingleSession)) {
+                        tuitionList.add(new Tuition(currStudent, currSession, i, j));
+                    }
+                }
             }
         }
     }
 
+    /**
+     * Returns true if dateOfSingleSession is between today and the day after tomorrow.
+     */
+    private boolean isWithinThreeDaysRange(SessionDate today, SessionDate dayAfterTomorrow,
```

Comment:
If this function checks for whether it's within today, tomorrow and the following day, do we need to pass in the parameters `today` and `dayAfterTomorrow`? cuz probably can use `LocalDateTime.now()` and `LocalDateTime.now().plusDays(2)`.

Unless this is generalised to a "check within data range" kinda function.

Reply:
hmmm, but i'm kind of using these variables above as well.. if i create them in the method then I'll have to create them twice..

Reply:
right... i see where you're coming from. i doubt there's an issue with code quality, just that i personally feel that it's abit odd to be named `isWithinThreeDaysRange` and we have to indicate today+2days (`dayAfterTomorrow`) date as well haha

if you're ok with it then feel free to merge!

on second thought: MUST the `dayAfterTomorrow` be today+2days? will it break if another dev inputs a random date? maybe assertion will be good here then.